listeners.go

```diff
@@ -289,8 +289,10 @@ func (na NetworkAddress) String() string {
 	return JoinNetworkAddress(na.Network, na.Host, na.port())
 }
 
-func isUnixNetwork(netw string) bool {
-	return netw == "unix" || netw == "unixgram" || netw == "unixpacket"
+// IsUnixNetwork returns true if the netw is a unix network.
+func IsUnixNetwork(netw string) bool {
+	return netw == "unix" || netw == "unixgram" || netw == "unixpacket" ||
+		netw == "unix+h2c" || netw == "unixgram+h2c" || netw == "unixpacket+h2c"
```

Comment:
I'm not sure how I feel about these... the `+h2c` thing is really a `reverseproxy`-specific thing, I'm not sure we need to have this in `listener.go` :thinking:

Reply:
Could we just do `strings.HasPrefix(netw, "unix")` ? Is that too permissive?

I'll let @mholt decide

Reply:
It is actually needed because `SplitNetworkAddress` otherwise does not return everything after "unix/" which is used in the new code.
The alternative would be some duplicated code, which I tried to avoid.

But let's hear from @mholt :)

Reply:
I need some more time to think about this solution -- I love this change overall though. I will review this for 2.6.3

Reply:
After "some more time" (4 months, haha sorry) I think that this is fine (for now); we could also do the `HasPrefix()` as that would be easier to maintain, I think.

Reply:
I pushed the `HasPrefix` change.